### ISSUE TITLE
Make it a package

### ISSUE THREAD

**Issue by pkrog** [workflow4metabolomics/easyrlibrary-lib#2]
Hi @melpetera ,

Maybe we could transform this repos into an R package that could be installed directly from GitHub, using devtools::install_github(). We could let the R files as they are today, since they are independent and don't use other library, I think, so it would still be possible to source the files as you do today.
However, converting into a package, would ease the testing and documentation, and allow anyone to use it as a package.
What do you think?

**Comment by lecorguille** [comment]
Is it mean that the repo will have to be pulled at runtime for each single job?

**Comment by pkrog** [comment]
No, that means that it will be possible to install it as a regular R package on the Galaxy server. Except that instead of installing from CRAN, you'll have to to install it from GitHub with the command:
```r
devtools::install_github('https://github.com/workflow4metabolomics/easyrlibrary-lib')
```
However I don't know if this is possible to specify its installation inside the requirements of an XML tool. Can we define a conda rule that installs an R package from GitHub?

**Comment by lecorguille** [comment]
Yes, I did so for the [r-probmetab](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/r-probmetab/1.1/meta.yaml) package
If so, we should set some releases.

**Comment by melpetera** [comment]
I am soooooo late but here is my plan:

- [x] Solving remaining PR 
- [ ] Creating a package tree structure and moving current files to the right place in it
- [ ] Adding mandatory package files (man, description, license...)
- [ ] Checking the package works
- [ ] Adding minor utilitary functions from other W4M tools (I am thinking about small codes from table_merge, ID_choice, Generic_filter...)
- [ ] Checking the functions are R4+ compliant (scripts are old)
- [ ] Take stock of the situation

Then the plan will be to see whether the package is ready to be used as a first version, and if yes whether we should aim for a CRAN deposit or if it is ok to construct the conda recipe directly from the GitHub repository.
